apps/web/src/ApplicationReadyGuard.tsx

```diff
@@ -1,25 +1,64 @@
 import { Navigate, useLocation } from 'react-router-dom';
 import { type PropsWithChildren, useLayoutEffect } from 'react';
-import { useAuth, useEnvironment } from './hooks';
-import { isStudioRoute } from './studio/utils/routing';
+import { useAuth, useEnvironment, useMonitoring, useRoutes } from './hooks';
+import { ROUTES } from './constants/routes';
+import { IS_EE_AUTH_ENABLED } from './config/index';
 
 export function ApplicationReadyGuard({ children }: PropsWithChildren<{}>) {
+  useMonitoring();
   const location = useLocation();
-  const { isLoading: isLoadingAuth, inPublicRoute } = useAuth();
-  const { isLoading: isLoadingEnvironment } = useEnvironment();
+  const { inPublicRoute, inStudioRoute } = useRoutes();
+  const { isUserLoaded, isOrganizationLoaded, currentUser, currentOrganization } = useAuth();
+  const { isLoaded: isEnvironmentLoaded } = useEnvironment();
 
-  const isLoading = isStudioRoute(location.pathname) ? isLoadingAuth : isLoadingAuth && isLoadingEnvironment;
+  const isLoaded = isUserLoaded && isOrganizationLoaded && isEnvironmentLoaded;
 
-  // Clean up the skeleton loader when the app is ready
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** This comment is still relevant‏